var/spack/repos/builtin/packages/py-cwl-utils/package.py

```diff
@@ -16,14 +16,27 @@ class PyCwlUtils(PythonPackage):
 
     license("Apache-2.0")
 
+    version("0.37", sha256="7b69c948f8593fdf44b44852bd8ef94c666736ce0ac12cf6e66e2a72ad16a773")
     version("0.21", sha256="583f05010f7572f3a69310325472ccb6efc2db7f43dc6428d03552e0ffcbaaf9")
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.8:", when="@0.29:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
 
     depends_on("py-cwl-upgrader@1.2.3:", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"))
     depends_on("py-rdflib", type=("build", "run"))
     depends_on("py-requests", type=("build", "run"))
     depends_on("py-cachecontrol", type=("build", "run"))
-    depends_on("py-schema-salad@8.3.20220825114525:8", type=("build", "run"))
+    depends_on("py-schema-salad@8.3.20220825114525:8", when="@:0.31", type=("build", "run"))
+    depends_on("py-schema-salad@8.8.20250205075315:8", when="@0.37", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.17.6:0.18", when="@0.30:0.37", type=("build", "run"))
```

Comment:
Can this also be an open ended range? Or was this removed already since 0.37?
```suggestion
    depends_on("py-ruamel-yaml@0.17.6:0.18", when="@0.30:", type=("build", "run"))
```